pkg/xds/server/components.go

```diff
@@ -263,18 +273,19 @@ func DefaultDataplaneSyncTracker(rt core_runtime.Runtime, reconciler, ingressRec
 					return ingressReconciler.Reconcile(envoyCtx, &proxy)
 				}
 
-				snapshotHash, err := meshSnapshotCache.GetHash(ctx, dataplane.GetMeta().GetMesh())
-				if err != nil {
-					return err
-				}
-				if prevHash != "" && snapshotHash == prevHash {
+				// if previous reconciliation was without an error AND current hash is equal to previous hash
+				// then we don't reconcile
+				if prevError == nil && prevHash != "" && snapshotHash == prevHash {
 					return nil
 				}
 				log.V(1).Info("snapshot hash updated, reconcile", "prev", prevHash, "current", snapshotHash)
 				prevHash = snapshotHash
 
-				mesh := core_mesh.NewMeshResource()
-				if err := rt.ReadOnlyResourceManager().Get(ctx, mesh, core_store.GetByKey(proxyID.Mesh, core_model.NoMesh)); err != nil {
+				// hacky way to be sure that Cluster Load Assignment is up to date
```

Comment:
I think this can better explained. Why it wasn't up to date.

TBH, on the first look - this cache invalidation will be almost the same as we remove the cache altogether. If we want to do it this way, we should immediately work on moving EDS to LinearCache...
the alternative - build better invalidation based on the hash.